.travis.yml

```diff
@@ -50,62 +43,91 @@ cache:
     - $HOME/.gradle/wrapper/
     - ./visualization/node_modules/nwjs-builder-phoenix/caches/
 
-deploy:
-  # releases the built packages on github when a commit is tagged
-  - provider: releases
-    api_key: $GITHUB_TOKEN
-    file_glob: true
-    file:
-      - "visualization/dist/packages/*.zip"
-      - "analysis/build/distributions/*.tar"
-    skip_cleanup: true
-    on:
-      tags: true
-      branch: master
-
-  # commits the gh-pages directory to the gh-pages branch. Does not ignore built ressources
-  - provider: pages
-    github_token: $GITHUB_TOKEN
-    skip_cleanup: true
-    local_dir: "gh-pages"
-    on:
-      tags: true
-      branch: master
-
-  # publishes visualization sonar results
-  - provider: script
-    script: visualization/script/sonar-publish.sh
-    skip_cleanup: true
-    on:
-      branch: master
-
-  # publishes analysis sonar results
-  - provider: script
-    script: analysis/script/sonar-publish.sh
-    skip_cleanup: true
-    on:
-      branch: master
-
-  # publishes visualization npm package
-  - provider: script
-    script: visualization/script/npm-publish.sh
-    skip_cleanup: true
-    on:
-      tags: true
-      branch: master
-
-  # publishes analysis npm package
-  - provider: script
-    script: analysis/script/npm-publish.sh
-    skip_cleanup: true
-    on:
-      tags: true
-      branch: master
-
-  # publishes visualization docker image
-  - provider: script
-    script: visualization/script/docker-publish.sh
-    skip_cleanup: true
-    on:
-      tags: true
-      branch: master
+jobs:
+  fast_finish: true
+  include:
+    - stage: test
+      name: "Visualization Unit Test and Linter"
+      before_script: cd visualization
```

Comment:
We have to run `npm install` here inside the `before_script` and below for other jobs